plugins/package-managers/node/src/main/kotlin/PackageJson.kt

```diff
@@ -141,11 +145,23 @@ data class PackageJson(
 
 private object AuthorListSerializer : JsonTransformingSerializer<List<Author>>(serializer<List<Author>>()) {
     override fun transformDeserialize(element: JsonElement): JsonElement =
-        // TODO: The string from the JSON primitive could be parsed into the dedicated author properties.
         when (element) {
             is JsonObject -> JsonArray(listOf(element))
-            is JsonPrimitive -> JsonArray(listOf(element.wrapPrimitiveInObject("name")))
-            is JsonArray -> JsonArray(element.map { it.wrapPrimitiveInObject("name") })
+            is JsonPrimitive -> JsonArray(listOf(element.toAuthorObject()))
+            is JsonArray -> JsonArray(element.map { it.toAuthorObject() })
+        }
+
+    private fun JsonElement.toAuthorObject(): JsonElement =
+        when (this) {
+            is JsonObject -> this
+
+            is JsonPrimitive -> {
+                val (name, email, url) = parseAuthorString(jsonPrimitive.contentOrNull)
+                val author = Author(name ?: content, email, url)
```

Comment:
Is `name ?: content` obsolete with the changes in `parseAuthorString()` ?

Reply:
Technically it would not be required as `name` should as a fallback also contain the full `content`. But `name` is nullable, and using `?: content` seems to be the better approach than `.orEmpty()` or anything else to make it non-nullable.

Reply:
Makes sense.

nit: `jsonPrimitive` might be replaced with `this`, but I'm ok to keep it as-is.